src/components/Button/index.js

```diff
@@ -254,47 +258,41 @@ class Button extends Component {
                 onPressOut={this.props.onPressOut}
                 onMouseDown={this.props.onMouseDown}
                 disabled={this.props.isLoading || this.props.isDisabled}
-                style={[
+                wrapperStyle={[
                     this.props.isDisabled ? {...styles.cursorDisabled, ...styles.noSelect} : {},
+                    (this.props.isDisabled && (this.props.success || this.props.danger)) ? styles.buttonOpacityDisabled : undefined,
+                    (this.props.isDisabled && !this.props.danger && !this.props.success) ? styles.buttonDisabled : undefined,
                     styles.buttonContainer,
                     this.props.shouldRemoveRightBorderRadius ? styles.noRightBorderRadius : undefined,
                     this.props.shouldRemoveLeftBorderRadius ? styles.noLeftBorderRadius : undefined,
                     ...StyleUtils.parseStyleAsArray(this.props.style),
                 ]}
+                style={({isHovered, isDisabled}) => [
+                    styles.button,
+                    this.props.small ? styles.buttonSmall : undefined,
+                    this.props.medium ? styles.buttonMedium : undefined,
+                    this.props.large ? styles.buttonLarge : undefined,
+                    this.props.success ? styles.buttonSuccess : undefined,
+                    this.props.danger ? styles.buttonDanger : undefined,
+                    (this.props.success && isHovered && !isDisabled) ? styles.buttonSuccessHovered : undefined,
+                    (this.props.danger && isHovered && !isDisabled) ? styles.buttonDangerHovered : undefined,
```

Comment:
The style here is not correct. Previously we had `const activeAndHovered = !this.props.isDisabled && hovered;` where the active state was solely base on the `isDisabled` prop. Now it's based on the `isDisabled` state which is based on both `isDisabled` prop and `isLoading` prop.

Reply:
you're right for both the above comments, changes applied.